### PR TITLE
DEV-2981: Revise SwagEditor

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Revises the contributor benefits (aka swag) editor.

#### Why are we doing this? How does it help us?

Improves UX, continues TypeScript migration.

#### How should this be manually tested? Please include detailed step-by-step instructions.

1. Edit a contribution page and add a new contributor benefits block.
2. Confirm that the block comes configured with a threshold of $240 and no swag options.
3. Confirm that the Update button is disabled in this state.
4. Add swag options and a name for the group in the text field.
5. Save changes.
6. Confirm that the published page shows the swag block as you configured it, and choices you made are saved after a contribution.
7. Confirm that the swag editor will not let you remove the last swag option.
8. Confirm that the swag editor will not let you add a duplicate swag option.
9. Confirm that the swag editor will not let you add a swag option longer than 40 characters.
10. Confirm that the swag editor will not let you change the threshold amount to a negative number.
11. Confirm that the swag editor will not let you change the threshold amount to a number with too many decimal places (e.g. 1234). 
12. Confirm that the swag editor will not let you enter a non-numeric threshold amount.
13. Confirm that the "opt-out by default" checkbox in the editor causes the checkbox on the published page to start either checked or unchecked.
14. Confirm that the complimentary NYT sub checkbox is present only if this option is enabled for the page in Django admin.
15. Confirm that checking the complimentary NYT sub checkbox in the editor shows this checkbox in the published page, and that the choice you make while making a contribution is saved.
16. Confirm that unchecked the NYT sub checkbox causes the checkbox to disappear from the published page, and that contributions are not marked as requesting an NYT sub.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

Updated screenshots.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

Updated docs perhaps.

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2981

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

File: spa/src/components/pageEditor/elementEditors/swag/SwagEditor.test.tsx [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+  });
+
+  it('disables updates if there is no swag group name defined', () => {
+    const setUpdateDisabled = jest.fn();
+
+    tree({ setUpdateDisabled, elementContent: { ...elementContent, swags: [{ swagOptions: ['one'] }] as any } });
+    expect(setUpdateDisabled.mock.calls).toEqual([[true]]);
+  });
+
+  it.todo('blocks updates if there are no swag options set');
```
Comment (GuiMend) [inline spa/src/components/pageEditor/elementEditors/swag/SwagEditor.test.tsx:66]: TODO

File: spa/src/components/pageEditor/elementEditors/swag/SwagEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+};
+
+export type SwagEditorProps = ElementDetailEditorProps<SwagElement['content']>;
+
+export function SwagEditor({
+  elementContent,
+  onChangeElementContent,
+  pagePreview,
+  setUpdateDisabled
+}: SwagEditorProps) {
```
Comment (GuiMend) [inline spa/src/components/pageEditor/elementEditors/swag/SwagEditor.tsx:31]: `setUpdateDisabled` is missing inside `SwagEditorProps` and also missing in storybook. That's why it's breaking

**Review by GuiMend — CHANGES_REQUESTED** [review]
SwagEditor story is broken:
Screenshot 2023-03-01 at 11 47 39

File: spa/src/components/pageEditor/elementEditors/swag/SwagEditor.test.tsx (hunk 1 above)
Comment (nrh-cklimas) [reply spa/src/components/pageEditor/elementEditors/swag/SwagEditor.test.tsx]: Fixed in b0decb40feb9d30f2401ab1a60028365b646d76b

File: spa/src/components/pageEditor/elementEditors/swag/SwagEditor.tsx (hunk 2 above)
Comment (nrh-cklimas) [reply spa/src/components/pageEditor/elementEditors/swag/SwagEditor.tsx:31]: Fixed in 00cd0c610cfe4e8c77684a52bd89a5fd9c46309b. This prop gets defined by `ElementDetailEditorProps` in [ElementEditor.types.ts](https://github.com/newsrevenuehub/rev-engine/blob/main/spa/src/components/pageEditor/editInterface/ElementEditor.types.ts).

**Review by GuiMend — APPROVED** [review]

**Review by GuiMend — APPROVED** [review]
👍

**Pull request closed without merging**